vevestaX/vevesta.py

```diff
@@ -201,93 +204,102 @@ def dump(self, techniqueUsed, filename=None, message=None, version=None, showMes
-        workbbok.save(filename)
-        # workbook.close()
-    
-    # to save the plot in a directory
-    def directorysave(fig, column):
-        script_dir = os.path.dirname('vevestaXdump')
-        results_dir = os.path.join(script_dir, '/')
-        sample_file_name = column
-
-        if not os.path.isdir(results_dir):
-            os.makedirs(results_dir)
-
-        fig.savefig(results_dir + sample_file_name)
-
-    def commit(self, techniqueUsed, filename=None, message=None, version=None, project_id=None):
+        directoryToDumpData = 'vevestaXDump'
+        self.__truncateFolder(directoryToDumpData)
+        # creating a new folder in current directory
+        Path(directoryToDumpData).mkdir(parents=True, exist_ok=True)
+
+        # checks if file exist then only loads it and create a new sheet for plots
+        if (os.path.isfile(fileName)):
+            workBook = openpyxl.load_workbook(fileName)
+            workBook.create_sheet('performancePlots')
+            plotSheet=workBook['performancePlots']
+            xAxis = list(nonNumericColumns['timestamp'])
+            columnValue = 2
+
+            for column in modelingData.columns:
+                yAxis = list(modelingData[column])
+
+                imageName = str(column)+'.png'
+                columnText = 'B'
+                columnText += str(columnValue)
+
+                # creates a seperate plots for every timestamp vs column and saves it
+                fig,ax=plt.subplots()
+                ax.plot(xAxis,yAxis, linestyle='-', marker='o')
+                # rotating the x axis labels
+                plt.xticks(rotation = 45)
+                plt.title('Timestamp vs '+str(column))
+                plt.xlabel('Timestamp')
+                plt.ylabel(str(column))
+
+                plt.savefig(os.path.join(directoryToDumpData,imageName), bbox_inches='tight') 
+                plt.close()
+
+                img = openpyxl.drawing.image.Image(os.path.join(directoryToDumpData,imageName))  
+                img.anchor = columnText
+                plotSheet.add_image(img)
+                
+
+                columnValue+=20
+
+            workBook.save(fileName)
+        workBook.close()
+
+    # to truncate the content inside the vevestaXDump folder if it exist
+    def __truncateFolder(self, folderName):
+        if os.path.isdir(folderName):
+            for filename in os.listdir(folderName):
+                file_path = os.path.join(folderName, filename)
+                try:
+                    if os.path.isfile(file_path) or os.path.islink(file_path):
+                        os.unlink(file_path)
+                    elif os.path.isdir(file_path):
+                        shutil.rmtree(file_path)
+                except Exception as e:
+                    print('Failed to delete %s. Reason: %s' % (file_path, e))
+
+    # generic function to get modelling sheet data from any excel file with sheetName modelling
+    def getModellingSheetData(self, fileName):
```

Comment:
this is not a generic function use function signature __getExcelSheetData(self, fileName, sheetName)